### DISCUSSION TITLE
IP-MIB  atNetAddress  Parsing error

### DISCUSSION THREAD

**Discussion by wgchaoer** [prometheus/snmp_exporter#1166] [Q&A]
the result:    atNetAddress="1.192.168.10"  ...      The IP address should not be in this style, the correct one should be    192.168.10.1

——————————————————————————————————————————————————

 HELP atIfIndex The interface on which this entry's equivalence is effective - 1.3.6.1.2.1.3.1.1.1
 TYPE atIfIndex gauge
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="01:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="0B:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="0C:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="28:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="59:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="B7:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="C9:00:00:00:00:00"} 1
atIfIndex{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="DC:00:00:00:00:00"} 1
HELP atNetAddress The NetworkAddress (e.g., the IP address) corresponding to the media-dependent `physical' address. - 1.3.6.1.2.1.3.1.1.3
TYPE atNetAddress gauge
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="01:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="0B:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="0C:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="28:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="59:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="B7:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="C9:00:00:00:00:00"} 1
atNetAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="DC:00:00:00:00:00"} 1
 HELP atPhysAddress The media-dependent `physical' address - 1.3.6.1.2.1.3.1.1.2
 TYPE atPhysAddress gauge
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="01:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="0B:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="0C:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="28:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="59:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="B7:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="C9:00:00:00:00:00"} 1
atPhysAddress{atIfIndex="2",atNetAddress="1.192.168.10",atPhysAddress="DC:00:00:00:00:00"} 1
————————————————————————————————————————————————————

my config :

![image](https://github.com/prometheus/snmp_exporter/assets/96378503/da772480-4680-4ed2-8a72-6297f5603cda)

**Comment by SuperQ** [comment]
This is `atTable` from `RFC1213-MIB`. Have you tried using the generator?

**Reply by SuperQ** [reply]
Related, you might want to look at the more modern `IP-MIB` `ipNetToPhysicalTable` (`1.3.6.1.2.1.4.35`). This table supports IPv6 and some additional details.
